File: Algorithms/DFS.cpp
struct TreeNode {
    int val;
    TreeNode* left;
    TreeNode* right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode* left, TreeNode* right) : val(x), left(left), right(right) {}
};

vector<string> DFS(TreeNode* tree) {
    stack<TreeNode*> open;
    vector<string> close;

    open.push(tree);
    while (!open.empty()) {
        TreeNode* temp = open.top();
        open.pop();
        if (temp == nullptr) close.emplace_back("null");
        else {
            close.emplace_back(to_string(temp->val));
            open.push(temp->right);
            open.push(temp->left);
        }
    }
    return close;
}

File: Algorithms/DFS.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


def DFS(tree: TreeNode) -> List:
    open = []
    close = []

    open.append(tree)
    while len(open) != 0:
        temp = open.pop()
        if temp is None:
            close.append("null")
        else:
            close.append(temp.val)
            open.append(temp.right)
            open.append(temp.left)
    return close

